twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php

```diff
@@ -374,14 +374,32 @@ public function isTwoFactorAuthEnabledForUser(IUser $user)
                 // We can exclude groups from the 2FA
                 $piexcludegroupsCSV = str_replace("|", ",", $piexcludegroups);
                 $groups = explode(",", $piexcludegroupsCSV);
+                $checkEnabled;
                 foreach($groups as $group) {
-                    if (($this->groupManager->isInGroup($user->getUID(), trim($group)) && $piexclude === "1")
-                    || (!($this->groupManager->isInGroup($user->getUID(), trim($group))) && $piexclude === "0")) {
-                        // The user is a group, that is allowed to pass with 1FA
-	                    $this->log("debug", "[isTwoFactorAuthEnabledForUser] User does not need 2FA");
-                        return false;
-                    };
+                	if($this->groupManager->isInGroup($user->getUID(), trim($group))) {
+		                $this->log("debug", "[isTwoFactorEnabledForUser] The user " . $user->getUID() . " is in group " . $group . ".");
+                		if($piexclude === "1"){
+			                $this->log("debug", "[isTwoFactorEnabledForUser] The group " . $group . " is excluded (User does not need 2FA).");
+                			return false;
+		                }
+		                if($piexclude === "0") {
+			                $this->log("debug", "[isTwoFactorEnabledForUser] The group " . $group . " is included (User needs 2FA).");
+			                return true;
+		                }
+	                }
+	                $this->log("debug", "[isTwoFactorEnabledForUser] The user " . $user->getUID() . " is not in group " . $group . ".");
+                	if($piexclude === "1"){
+		                $this->log("debug", "[isTwoFactorEnabledForUser] The group " . $group . " is excluded (User may not need 2FA).");
```

Comment:
Wait. We are doing a exclude-groups. The user is not in the excludes up til now. So it could be, that the user **needs** 2FA. Your text *(User may not need 2FA)* is wrong.
Also the return value "true" is prepared, which would mean, that the user needs 2FA (which is correct).